Symfony/CS/Tests/FixerFactoryTest.php

```diff
@@ -279,4 +279,39 @@ function ($name) {
 
         return $cases;
     }
+
+    public function testHasRule()
+    {
+        $factory = new FixerFactory();
+        $factory->registerBuiltInFixers();
+
+        $f1 = $this->getMock('Symfony\CS\FixerInterface');
+        $f1->expects($this->any())->method('getName')->willReturn('f1');
+        $f2 = $this->getMock('Symfony\CS\FixerInterface');
+        $f2->expects($this->any())->method('getName')->willReturn('f2');
+        $f3 = $this->getMock('Symfony\CS\FixerInterface');
+        $f3->expects($this->any())->method('getName')->willReturn('f3');
+        $factory->registerFixer($f1);
+        $factory->registerCustomFixers(array($f2, $f3));
+
+        $this->assertTrue($factory->hasRule('f1'), 'Should have f1 fixer');
+        $this->assertTrue($factory->hasRule('f2'), 'Should have f2 fixer');
+        $this->assertTrue($factory->hasRule('f3'), 'Should have f3 fixer');
+        $this->assertFalse($factory->hasRule('dummy'), 'Should not have dummy fixer');
+    }
+
+    public function testHasRuleWithChangedRuleSet()
+    {
+        $factory = new FixerFactory();
+        $factory->registerBuiltInFixers();
+
+        $f1 = $this->getMock('Symfony\CS\FixerInterface');
+        $f1->expects($this->any())->method('getName')->willReturn('f1');
+        $factory->registerFixer($f1);
+
+        $this->assertTrue($factory->hasRule('f1'), 'Should have f1 fixer');
+
+        $factory->useRuleSet(new RuleSet(array('@PSR1' => true)));
```

Comment:
If we remove `@PSR1` set then this will throw an exception.
Use mocks to create f1 and f2, then create ruleset without one of them.